lib/component-data/actions.js

```diff
@@ -37,15 +38,12 @@ function logSaveError(uri, e, store) {
  * @return {Promise}
  */
 function clientSave(uri, data, oldData, store, {eventID, snapshot, paths}) { // eslint-disable-line
-  store.commit(CURRENTLY_SAVING, true);
-  return modelSave(uri, data, {eventID, snapshot})
+  return modelSave(uri, data)
     .catch((e) => {
       // if model.js errors (or times out), re-render component with old data
       // and show the end user an error message
       logSaveError(uri, e, store);
-      return Promise.reject(store.dispatch('render', { uri, data: oldData, snapshot, paths }).then(() => {
-        store.commit(CURRENTLY_SAVING, false);
-      })); // don't continue saving
+      return Promise.reject(store.dispatch('render', { uri, data: oldData, snapshot, paths })); // don't continue saving
```

Comment:
Here `reject` is being called with the result of a dispatch, which doesn't seem like the same thing as a *reason*, which is the expected argument to `reject`: http://www.ecma-international.org/ecma-262/6.0/#sec-promise-reject-functions

It should be possible to render the old data and then reject the promise with a reason.